dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/layers/ConvolutionalLayer.java

```diff
@@ -0,0 +1,108 @@
+   * This layer works for only 2D inputs.
+   * In a forward pass,
+   * computes the product between weight and the input within kernel range and produce activation matrix.
+   * In a backward pass,
+   * error of each input pixel comes from
+   * the product between weight and errors of output pixels affected by the input pixel in feedforward step.
+   */
+public final class ConvolutionalLayer extends LayerBase {
+
+  private final int[] outputShape;
+  private final int paddingHeight;
+  private final int paddingWidth;
+  private final int strideHeight;
+  private final int strideWidth;
+  private final int kernelHeight;
+  private final int kernelWidth;
+
+  @Inject
+  private ConvolutionalLayer(@Parameter(LayerIndex.class) final int index,
+                             @Parameter(LayerInputShape.class) final String inputShape,
+                             @Parameter(StrideHeight.class) final int paddingHeight,
+                             @Parameter(StrideWidth.class) final int paddingWidth,
+                             @Parameter(StrideHeight.class) final int strideHeight,
+                             @Parameter(StrideWidth.class) final int strideWidth,
+                             @Parameter(KernelHeight.class) final int kernelHeight,
+                             @Parameter(KernelWidth.class) final int kernelWidth,
+                             final LayerParameterInitializer layerParameterInitializer) {
+    super(index, inputShape);
+    this.paddingHeight = paddingHeight;
+    this.paddingWidth = paddingWidth;
+    this.strideHeight = strideHeight;
+    this.strideWidth = strideWidth;
+    this.kernelHeight = kernelHeight;
+    this.kernelWidth = kernelWidth;
+    this.outputShape = layerParameterInitializer.getOutputShape();
+    setLayerParameter(layerParameterInitializer.generateInitialParameter());
+  }
+
+  @Override
+  public int[] getOutputShape() {
+    return outputShape;
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public boolean isLearnable() {
+    return false;
+  }
+
+  /**
+   * Computes output values for this convolutional layer.
+   * @param input input values for this layer.
+   * @return output values for this layer.
+   */
+  @Override
+  public Matrix feedForward(final Matrix input) {
+    throw new RuntimeException("Not Implemented");
+  }
+
+  /**
+   * Computes errors for this convolutional layer.
+   * @param input the input values for this layer.
+   * @param activation the output values.
+   * @param nextError the errors of the next layer - the one closer to the output layer.
+   * @return errors for this layer with the specified input value.
+   */
+  @Override
+  public Matrix backPropagate(final Matrix input, final Matrix activation, final Matrix nextError) {
+    throw new RuntimeException("Not Implemented");
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public LayerParameter generateParameterGradient(final Matrix input, final Matrix error) {
+    return LayerParameter.newBuilder()
```

Comment:
The method to generate `ConvolutionalLayer`'s parameter gradient is different from `FullyConnectedLayer`'s one, isn't it? I referred to [this webpage](http://ufldl.stanford.edu/tutorial/supervised/ConvolutionalNeuralNetwork).